de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/process/M_Inventory_CreateCountLinesFromHU.java

```diff
@@ -117,35 +135,55 @@ private I_M_Inventory getInventory()
 				.iterateAndStream();
 	}
 
-	private Stream<I_M_InventoryLine> createInventoryLines(final I_M_HU hu)
+	private Stream<I_M_InventoryLine> createUpdateInventoryLines(@NonNull final I_M_HU hu, final boolean updateLine)
 	{
 		return Services.get(IHandlingUnitsBL.class)
 				.getStorageFactory()
 				.streamHUProductStorages(hu)
-				.map(huProductStorage -> createInventoryLine(huProductStorage));
+				.map(huProductStorage -> createUpdateInventoryLine(huProductStorage, updateLine));
 	}
 
-	private I_M_InventoryLine createInventoryLine(final IHUProductStorage huProductStorage)
+	private I_M_InventoryLine createUpdateInventoryLine(@NonNull final IHUProductStorage huProductStorage, final boolean updateLine)
 	{
 		final I_M_Inventory inventory = getInventory();
-		final I_M_InventoryLine inventoryLine = InterfaceWrapperHelper.newInstance(I_M_InventoryLine.class);
-		inventoryLine.setM_Inventory(inventory);
-		inventoryLine.setAD_Org_ID(inventory.getAD_Org_ID());
-		inventoryLine.setM_Product_ID(huProductStorage.getM_Product_ID());
-		inventoryLine.setM_AttributeSetInstance_ID(0);
-
+		final I_M_InventoryLine inventoryLine;
 		final I_M_HU hu = huProductStorage.getM_HU();
-		inventoryLine.setM_Locator_ID(hu.getM_Locator_ID());
-		inventoryLine.setM_HU_ID(hu.getM_HU_ID());
-		inventoryLine.setM_HU_PI_Item_Product(null); // TODO
-		inventoryLine.setQtyTU(BigDecimal.ZERO); // TODO
+		if (updateLine)
+		{
+			inventoryLine = fetchInventoryLineForHU(hu);
+		}
+		else
+		{
+			inventoryLine = InterfaceWrapperHelper.newInstance(I_M_InventoryLine.class);
+			inventoryLine.setM_Inventory(inventory);
+			inventoryLine.setAD_Org_ID(inventory.getAD_Org_ID());
+			inventoryLine.setM_Product_ID(huProductStorage.getM_Product_ID());
+			inventoryLine.setM_AttributeSetInstance_ID(0);
+
+			inventoryLine.setM_Locator_ID(hu.getM_Locator_ID());
+			inventoryLine.setM_HU_ID(hu.getM_HU_ID());
+			inventoryLine.setM_HU_PI_Item_Product(null); // TODO
+			inventoryLine.setQtyTU(BigDecimal.ZERO); // TODO
+
+			inventoryLine.setC_UOM(huProductStorage.getC_UOM());
+		}
+
 
-		inventoryLine.setC_UOM(huProductStorage.getC_UOM());
 		inventoryLine.setQtyBook(huProductStorage.getQty());
 		inventoryLine.setQtyCount(huProductStorage.getQty());
 
 		InterfaceWrapperHelper.save(inventoryLine);
 
 		return inventoryLine;
 	}
+
+	private I_M_InventoryLine fetchInventoryLineForHU(final I_M_HU hu)
```

Comment:
generally, pls pls pls don't use models when simply the M_HU_ID would be enough!